soc/arm/nordic_nrf/nrf53/soc.c

```diff
@@ -96,7 +96,7 @@ static int nordicsemi_nrf53_init(const struct device *arg)
 		 + ((offset - 8) << 4) + 32) >> 6;
 
 	nrf_oscillators_hfxo_cap_set(NRF_OSCILLATORS, true, capvalue);
-#else
+#elif defined(CONFIG_SOC_HFXO_CAP_EXTERNAL)
```

Comment:
should there be an else with a build failure ?
Or is that a valid use-case.

Reply:
No, this is intentional. The else case here is the `SOC_HFXO_CAP_DEFAULT` choice, which by definition should do nothing.

Reply:
Great, thanks for explanation.
Just playing safe as an `else` changed into `elif` without a new `else` is easy to forget.

Reply:
We need an #else here for Misra C compliance